dip-0027.md

```diff
@@ -0,0 +1,163 @@
+
+#### Signing Asset Unlock Transactions
+
+An active Platform validator quorum quorum must sign Asset Unlock transactions per the process described in [DIP-7](./dip-0007.md). The signing session’s requestID is calculated as follows:
+
+    requestID = SHA256(SHA256("plwdtx", index))
+
+For reference, the following table shows some example index values and their request IDs:
+
+| **Index** | **Request ID** |
+|-----------|----------------|
+| 101       | fcc76a643c5c668244fdcef09833955d6f4b803fa6c459f7732983c2332389fd |
+| 123456789 | ebd9d75ad72184bec8e8f25a499eb5386ba564eb7ce70a4a3b9db652599d6d72 |
+
+The SignID is defined as _SignID = SHA256(quorumHash, requestID, sigHash)_ where sigHash is the hash of the entire transaction with the quorumSig field set to zeros. The transaction must be hashed as described in [DIP-2](./dip-0002.md#serialization-hashing-and-signing).
+
+### Client verification
+
+All clients must verify the quorumSig field before accepting the transaction. If the quorum is recent and the transaction is not yet mined, light clients should attempt to verify the quorumSig by retrieving the quorum public key as described in DIP-4 and verifying the signature. If the quorum is not recent and the transaction is not mined, the transaction should be ignored until it is mined to prevent attacks on the light client.
+
+### Withdrawal Completion
+
+Since Asset Unlock transactions do not have inputs, they are not eligible for InstantSend. Once mined into a block and subsequently ChainLocked, they are final and identical to all other outputs. A withdrawal should be considered complete when the corresponding Asset Unlock transaction is finalized on the Core chain.
+
+Asset Unlock transactions might not be mined for multiple reasons. For example, the quorum who signed a transaction expired, Core fees were too low, or the withdrawal limit was reached. To handle this situation, Asset Unlock transactions have an expiration period based on the quorum lifetime. Transactions are considered invalid if they are not signed by either the active quorum or the most recent inactive quorum. In this case, the issuer can retry the withdrawal. To ensure that expired transactions are not included in the Core chain, Asset Lock transactions are refused once the block height exceeds _signHeight_ by 48 or more (i.e., height > _signHeight_ + 48).
+
+## Withdrawal Safety
+
+As a way to avoid a catastrophic failure if Platform is compromised, Core will limit credit pool withdrawals for at least the first release of Platform. The withdrawal limits may be re-evaluated and updated periodically as the system matures. To evaluate withdrawal validity, withdrawal amounts from the last 576 blocks should be tallied. The withdrawal should not be mined if:
+
+* It requests more DASH than the credit pool contains
+* It requests more than 1000 DASH
```

Comment:
```suggestion
* The withdrawal would result in more than 1000 DASH reduction in the credit pool over the 576 block-window
```

Reply:
Thanks! I've included it - which reset your reveiw.